File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        int n = grid.size();
        std::vector<std::vector<int>> res(n - 2, std::vector<int>(n - 2));
        for (int i = 0; i < n - 2; ++i)
            for (int j = 0; j < n - 2; ++j)
                for (int ii = i; ii < i + 3; ++ii)
                    for (int jj = j; jj < j + 3; ++jj)
                        res[i][j] = std::max(res[i][j], grid[ii][jj]);
        return res;
    }
};

File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.rs
use std::cmp::max;

impl Solution {
    pub fn largest_local(grid: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let n = grid.len();
        let mut res = vec![vec![0 as i32;n - 2];n - 2];
        
        for i in 0..n - 2 {
            for j in 0..n - 2 {
                for ii in i..i + 3 {
                    for jj in j..j + 3 {
                        res[i][j] = res[i][j].max(grid[ii][jj]);
                    }
                }
            }
        }
        
        res
    }
}